### PR TITLE
[PLAT-11288] Create new CompleteNavigation component

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Create `CompleteNavigation` component, to be included in an application to end an open navigation span

## Testing

Unit tests that the appropriate method is called when a component is mounted or unmounted

### CODE REVIEW

File: packages/react-navigation/lib/complete-navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  
+  // @ts-expect-error
+  context: React.ContextType<typeof NavigationContext>;
+
+  constructor(props: Props, context: typeof NavigationContext) {
+    super(props);
+
+    // @ts-expect-error
+    context.blockNavigationEnd();
+  }
```
Comment (yousif-bugsnag) [inline packages/react-navigation/lib/complete-navigation.tsx:20]: I think these typescript errors can be fixed by the following?
https://legacy.reactjs.org/docs/context.html#classcontexttype

```suggestion
export class CompleteNavigation extends React.Component<Props> {
  static contextType = NavigationContext;
  
  declare context: React.ContextType<typeof NavigationContext>;

  constructor(props: Props) {
    super(props);

    this.context.blockNavigationEnd();
  }
```

Comment (yousif-bugsnag) [reply packages/react-navigation/lib/complete-navigation.tsx:20]: ah maybe not, that breaks the unit tests...

File: packages/react-navigation/test/utils/create-test-client.ts
Comment (yousif-bugsnag) [inline packages/react-navigation/test/utils/create-test-client.ts:1]: I'm a bit confused as to why we have this in addition to the mocked `@bugsnag/react-native-performance`?

File: packages/react-navigation/lib/complete-navigation.tsx (hunk 1 above)
Comment (gingerbenw) [reply packages/react-navigation/lib/complete-navigation.tsx]: updated!

File: packages/react-navigation/test/utils/create-test-client.ts
Comment (gingerbenw) [reply packages/react-navigation/test/utils/create-test-client.ts:1]: gone!

File: packages/test-utilities/lib/create-span.ts [hunk 2]
```diff
@@ -15,6 +16,15 @@ export function createSamplingProbability (rawProbability: number): SpanProbabil
   }
 }
 
+export function createSpan (name: string): Span {
```
Comment (yousif-bugsnag) [inline packages/test-utilities/lib/create-span.ts:19]: Can we remove the `name` param here as it's not being used?

File: packages/react-navigation/lib/complete-navigation.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+interface Props extends PropsWithChildren {
+  on: 'mount' | 'unmount';
+}
+
+/** End the current navigation span when the component is mounted, unmounted or the `on` prop is `true` */
+export class CompleteNavigation extends React.Component<Props> {
+  static contextType = NavigationContext;
+  
+  context!: React.ContextType<typeof NavigationContext>;
```
Comment (yousif-bugsnag) [inline packages/react-navigation/lib/complete-navigation.tsx:12]:
```suggestion
  declare context: React.ContextType<typeof NavigationContext>;
```

**Review by yousif-bugsnag — APPROVED** [review]
couple of minor suggestions, otherwise LGTM

File: packages/test-utilities/lib/create-span.ts (hunk 2 above)
Comment (gingerbenw) [reply packages/test-utilities/lib/create-span.ts:19]: It will get used in the next PR to make assertions on spans a little easier

**Pull request merged**